dask/array/tests/test_array_core.py

```diff
@@ -394,6 +402,9 @@ def test_elemwise_with_ndarrays():
     assert eq(b + x, x + y)
     assert eq(a + y, x + y)
     assert eq(y + a, x + y)
+    # Error on shape mismatch
+    assert raises(ValueError, lambda: a + y.T)
```

Comment:
it's still a little annoying to me that we don't have a setup for testing specific error message.... *sigh*